### DISCUSSION TITLE
mbsync plugin should ask before making any changes

### DISCUSSION THREAD

**Discussion by Maxr1998** [beetbox/beets#4247] [Ideas]
When running `beet mbsync`, the command should print the proposed changes, and then ask for approval, similarly to `beet modify`.

That's because changes on MusicBrainz may not always be expected or wanted.

Always running the command with `-p` first, checking the output, and running it again without the flag is tedious (and may even introduce race conditions if something gets edited in between, even if unlikely).

For the old behavior, a `-y` flag could be introduced, or a configuration file option.

**Comment by sampsyo** [comment]
Seems like a pretty solid idea! I'm not sure what the default should be—there's an argument to be made that it should go the other way for backwards compatibility, but defaulting to confirmation would unarguably be safer—but we can work that out in the implementation. Would you mind hitting that "Create issue from discussion" button so we can track this as a feature request?

**Reply by Maxr1998** [reply]
Sure thing!
When defaulting to no confirmation to keep backwards compatibility, I'd say a config option for the default (instead of only a CLI flag) would be useful.

But I'm personally in favor of making the safer option the default.
